### ISSUE TITLE
ECCC: Past forecast is localized but not future forecast

### ISSUE THREAD

**Issue by elydpg** [breezy-weather/breezy-weather#1139]
### Steps to reproduce

1. Make sure app language is set to something other than English (Spanish for testing purposes)
2. Add a location from ECCC expecting "Thunderstorm"
3. Check the forecast for when the thunderstorm are expected
4. Check the forecast for when something else is expected

### Expected behavior

The current weather condition in both locations should appear in the same language (either in the app language or English)

### Actual behavior

Thunderstorm is localized while the other condition is not

### Weather source used

ECCC

### Breezy Weather version

5.2.4 standard

### Android version

14

### Device

Pixel 8 Pro Emulated

### Other details

Applies to both the current weather condition and forecasts. Forecasts are easier to demo though, as below:

https://github.com/breezy-weather/breezy-weather/assets/8622264/876eea9b-ce38-4fe5-9c35-b5dce1bb7bcf

~~Demo is in Spanish, but French has this exact problem, which is surprising for ECCC which should support at least English and French...~~ EDIT: On Android 13+ French works fine for ECCC

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open OR closed issue.
- [X] I checked that my issue is not covered in the [Frequently Asked Questions](https://github.com/breezy-weather/breezy-weather/blob/main/HELP.md) document.
- [X] I have updated the app to **[version 5.2.4 or later](https://github.com/breezy-weather/breezy-weather/releases/latest)** and I can reproduce the issue in this version.
- [X] I wrote my request in English, and any screenshot I attached are also in English (language can be changed from Settings > Appearance), unless my issue only happens with a specific language. I can write in my native language under the English text, if necessary.
- [X] I understand that this app is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY, so I may not get a positive reply to my request, if any at all.

**Comment by papjul** [comment]
ECCC doesn't support Spanish. ECCC doesn't support past forecast.

**Comment by elydpg** [comment]
> ECCC doesn't support Spanish

Right? Shouldn't it show up in English then?

> ECCC doesn't support past forecast, we compute it ourselves.

Isn't this future/current forecast though? It can't be past forecast for a location that was just added... In any case, only the thunder is localized: In this example both the current forecast and future forecasts are English (probably as intended) even though the app language is in Spanish:

https://github.com/breezy-weather/breezy-weather/assets/8622264/6048ed15-31aa-4259-b9af-4a318f90cc9f

**Comment by elydpg** [comment]
Okay I've done some more testing and I found out a few things:

Firstly, French has no issues with ECCC; I was just mixing that up with the problems I was having on Android 12.

Second, I'm pretty sure that future forecasts are unaffected, but it's possible that in the Labrador city example there was some past data subbed in where normally future data would go. That is the forecast for "today" so I think that makes sense. Current forecasts are definitely affected though.

Third, it also seems to happen with "Clear sky": Here is another example with "Thunderstorm", and an example with "Clear sky":

https://github.com/breezy-weather/breezy-weather/assets/8622264/e4493be2-988f-4a61-b2b7-4cddb5db4737

https://github.com/breezy-weather/breezy-weather/assets/8622264/f75d9c54-a7de-4679-917e-4f7360466933

However, "Mostly cloudy", "Sunny", and "Mainly sunny" are only localized into English or French, as one would expect:

https://github.com/breezy-weather/breezy-weather/assets/8622264/8d97fcf1-91af-4c2f-ad2e-5dc603627eac

https://github.com/breezy-weather/breezy-weather/assets/8622264/bd052218-5a51-43a8-8ebc-776f44037e52

I think I might have figured out what is going on though: I checked against the official Environment Canada app in the "Thunderstorm" case, and it turns out there was no current weather condition provided:

![Screenshot_20240615-161944_WeatherCAN](https://github.com/breezy-weather/breezy-weather/assets/8622264/92c0436c-8a15-425a-a260-d17c6055cf9e)

So my guess is that the app is computing weather conditions whenever it finds it absent from source, and that is being localized, while the conditions from source are only being localized to English or French, as is intended for ECCC. This should theoretically affect any source that sends weather conditions only some of the time and supports fewer languages than the app does. Not sure what a good solution is though

**Comment by papjul** [comment]
> Not sure what a good solution is though

Solution to which issue exactly?

**Comment by elydpg** [comment]
That locally generated forecast names are localized even when the source doesn't support it

**Comment by papjul** [comment]
How is that an issue? It's the opposite, we take the opportunity that nothing is provided by the source to use our own translations to not leave it blank, or force to translate it only in a language supported by the source, which goes against the language chosen everywhere else for the UI.

**Comment by papjul** [comment]
> I suppose it's a consistency issue

In that case, to ensure consistency, I suggest you [contact ECCC](https://www.canada.ca/en/environment-climate-change/corporate/contact.html) to let them know that Spanish translation is important to you as a Canadian citizen and you would like them to provide Spanish translations in their API.

Please re-open an issue only once it is implemented on their side so that we can add support for it on Breezy Weather side.

**Comment by elydpg** [comment]
I would argue theb that the issue is moreso then ECCC not providing current weather conditions rather than with compatibilities with specific languages, but since the app seems to be doing what it's supposed to do in the absence of weather condition data from source I don't see any need to keep this issue open

**Comment by papjul** [comment]
Yes, you can also suggest them to add weather text in current.
